src/CMakeLists.txt

```diff
@@ -0,0 +1,96 @@
+find_package(Freetype REQUIRED)
+find_package(glad CONFIG REQUIRED)
+find_package(glm CONFIG REQUIRED)
+find_package(SDL2 CONFIG REQUIRED)
+find_package(sdl2-image CONFIG REQUIRED)
+find_package(sdl2-mixer CONFIG REQUIRED)
+find_package(spdlog CONFIG REQUIRED)
+
+# lua doesn't provide native cmake support
+find_path(LUA_INCLUDE_DIR lua.h)
+include_directories(${LUA_INCLUDE_DIR})
+
+find_library(LUA_LIBRARY lua)
+link_libraries(${LUA_LIBRARY})
+
+find_package(Doxygen QUIET)
+
+file(GLOB_RECURSE NOVELRT_LIB_HEADERS CONFIGURE_DEPENDS ../include/*.h)
+source_group(TREE ${CMAKE_SOURCE_DIR} FILES ${NOVELRT_LIB_HEADERS})
+
+file(GLOB_RECURSE NOVELRT_LIB_SOURCES CONFIGURE_DEPENDS NovelRT/*.cpp)
+source_group(TREE ${CMAKE_SOURCE_DIR} FILES ${NOVELRT_LIB_SOURCES})
+
+set(NOVELRT_LIB_LINK_LIBRARIES
+  Freetype::Freetype
+  glad::glad
+  glm
+  SDL2::SDL2-static
+  SDL2::SDL2_image
+  SDL2::SDL2_mixer
+  SDL2::SDL2main
+  spdlog::spdlog
+)
+
+if(NOT WIN32)
+  set(NOVELRT_LIB_LINK_LIBRARIES
+    ${NOVELRT_LIB_LINK_LIBRARIES}
+    m
+    stdc++fs
+  )
+endif()
+
+if (WIN32)
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
+
+add_library(NovelRT SHARED ${NOVELRT_LIB_HEADERS} ${NOVELRT_LIB_SOURCES})
```

Comment:
Unix automatically prepends `lib` to the target name, so we want just `NovelRT`. This gives us `NovelRT.dll`/`NovelRT.lib` on Windows and `libNovelRT.so`/libNovelRT.a` on Unix. We were getting `libNovelRT.dll`/`libNovelRT.lib` and `liblibNovelRT.so`/`liblibNovelRT.a` which I believe is undesirable.

Reply:
I can confirm as well - this was the output being delivered by Emscripten on WSL Ubuntu. Leaving here so Matt can review :)

Reply:
I am happy with this. Resolving.